File: trunk/emsi/asp-net-app/WebEMSOF--feature-track-proofs-of-payment/component/db/Class_db_payment_proof_methods.pas
unit Class_db_payment_proof_methods;

interface

uses
  Class_db;

type
  TClass_db_payment_proof_methods = class(TClass_db)
  private
    { Private Declarations }
  public
    constructor Create;
    procedure BindDropDownList(target: system.object);
  end;

implementation

uses
  borland.data.provider,
  system.web.ui.webcontrols;

constructor TClass_db_payment_proof_methods.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
end;

procedure TClass_db_payment_proof_methods.BindDropDownList(target: system.object);
var
  bdr: bdpdatareader;
begin
  self.Open;
  DropDownList(target).Items.Clear;
  DropDownList(target).Items.Add(listitem.Create('-- Select --',''));
  //
  bdr := Borland.Data.Provider.BdpCommand.Create
    ('SELECT code,description FROM payment_proof_method_code_description_map ORDER BY description',connection).ExecuteReader;
  while bdr.Read do begin
    DropDownList(target).Items.Add(listitem.Create(bdr['description'].tostring,bdr['code'].tostring));
  end;
  bdr.Close;
  self.Close;
end;

end.

File: trunk/emsi/asp-net-app/WebEMSOF/component/db/Class_db_payment_proof_methods.cs
using MySql.Data.MySqlClient;
using System;

using System.Web.UI.WebControls;
using Class_db;
namespace Class_db_payment_proof_methods
{
    public class TClass_db_payment_proof_methods: TClass_db
    {
        //Constructor  Create()
        public TClass_db_payment_proof_methods() : base()
        {
            // TODO: Add any constructor code here

        }
        public void BindListControl(object target)
        {
            MySqlDataReader dr;
            this.Open();
            ((target) as ListControl).Items.Clear();
            ((target) as ListControl).Items.Add(new ListItem("-- Select --", ""));
            dr = new MySqlCommand("SELECT code,description FROM payment_proof_method_code_description_map ORDER BY description", this.connection).ExecuteReader();
            while (dr.Read())
            {
                ((target) as ListControl).Items.Add(new ListItem(dr["description"].ToString(), dr["code"].ToString()));
            }
            dr.Close();
            this.Close();
        }

    } // end TClass_db_payment_proof_methods

}
